### ISSUE TITLE
setCookie ignores .path

### ISSUE THREAD

**Issue by edyu** [zigzap/zap#20]
When handling request path of multiple levels such as `/a/b/c` (my particular case was `/oauth2/google/redirect`, `setCookie` will automatically set the path to `/a/b`. On a subsequent visit, the cookie will only be available when you visit any path below `/a/b` so if you are visiting from a path such as `/` or `/a` the request handler won't see the cookie. This `may` be a security feature but there is no way to override it as `setCookie({.path = "/" })` will be ignored.

I tried using the following code
```
try r.setCookie(.{
    .name = self.settings.cookie_name,
    .value = state,
    .max_age_s = self.settings.cookie_maxage,
    .path = "/",
}) catch |err| {

```
Instead I have to use the following workaround to make it work:
```
const cookie_str = try std.fmt.allocPrint(self.allocator, "{s}={s};Path=/;Max-Age={d}", .{ 
    self.settings.cookie_name, state, self.settings.cookie_maxage 
});
defer self.allocator.free(cookie_str);
try r.setHeader("Set-Cookie", cookie_str);
```

**Comment by renerocksai** [comment]
There was a bug in facil.io where it would overwrite certain parts of the cookie header with spaces, resulting in `path` being converted to `ath`, for example. This is fixed in the latest version of zap's fork of facil.io. I'll release a new ZAP shortly.

**Comment by renerocksai** [comment]
fixed in 2885b2642b40bf10a5798af619e6338e070642ba
